File: leetcode/70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n,vector<int> &memo) {
        if(n<=2){
            return n;
        }
        if(memo[n]!=-1){
            return memo[n];
        }
        memo[n]=climbStairs(n-1,memo)+climbStairs(n-2,memo);
        return memo[n];
    }
    int climbStairs(int n){
        vector<int> memo(n+1,-1);
        return climbStairs(n,memo);
    }
};

File: leetcode/70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp=[-1]*(n+1)
        dp[0] = 1
        dp[1] = 1
        def helper(n,dp):
            if dp[n] != -1:
                return dp[n]
            dp[n] = helper(n-1,dp)+helper(n-2,dp)
            return dp[n]
        helper(n,dp)
        return dp[n]